siege_utilities/survey/weights.py

```
@@ -0,0 +1,82 @@
+    WeightingConvergenceError
+        If weightipy fails to converge.
+    ImportError
+        If weightipy is not installed (pip install siege-utilities[survey]).
+    """
+    try:
+        from weightipy import rake
+    except ImportError as exc:
+        raise ImportError(
+            "weightipy is required for RIM weighting. "
+            "Install it with: pip install siege-utilities[survey]"
+        ) from exc
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)